src/tools/stardb/startextdump.cpp

```diff
@@ -250,9 +250,30 @@ bool DumpStarDatabase(istream& in, ostream& out)
 
         out << catalogNum << ' ';
         out << setprecision(7);
-        out << x << ' ' << y << ' ' << z << ' ';
-        out << setprecision(4);
-        out << ((float) absMag / 256.0f) << ' ';
+        
+        if (spherical)
+        {
+            Eigen::Vector3d eclipticpos = {(double) x, (double) y, (double) z};
+            Eigen::Vector3d pos = astro::eclipticToEquatorial(eclipticpos);
+            double distance = sqrt(x * x + y * y + z * z);
+            double phi = -acos(pos.y() / distance) * 180 / PI;
```

Comment:
Why `-` before `acos`? If `y' = d cosφ` then `φ = acos y'/d`

Reply:
Well, the declination is always between -90° and 90°. When converting to φ you subtract 90° (https://en.wikibooks.org/wiki/Celestia/Binary_Star_File) so φ should always be between -180° and 0°. Problem is that acos outputs angles in the interval [0, π]. To make it work for [-π, 0] I added the negative sign.

Reply:
please leave then a comment why you use a minus